tests/backup/backup_restore_basic_test.go

```diff
@@ -3548,3 +3548,172 @@ var _ = Describe("{BackupNamespaceInNfsRestoredFromS3}", func() {
 		}
 	})
 })
+
+// DeleteS3ScheduleAndCreateNfsSchedule deletes s3 schedule and starts NFS schedule or vice-versa
+var _ = Describe("{DeleteS3ScheduleAndCreateNfsSchedule}", func() {
+	var (
+		s3CloudCredName          string
+		s3CloudCredUID           string
+		firstBkpLocationName     string
+		firstBackupLocationUID   string
+		secondBackupLocationName string
+		secondBackupLocationUID  string
+		schedulePolicyName       string
+		schedulePolicyUID        string
+		secondScheduleName       string
+		firstScheduleName        string
+		firstSchBackupName       string
+		scheduleUid              string
+		srcClusterUid            string
+		providers                []string
+		appNamespaces            []string
+		schedulePolicyInfo       *api.SchedulePolicyInfo
+		scheduledAppContexts     []*scheduler.Context
+		contexts                 []*scheduler.Context
+		appContexts              []*scheduler.Context
+		schedulePolicyInterval   = int64(15)
+	)
+	backupLocationMap := make(map[string]string)
+	JustBeforeEach(func() {
+		StartTorpedoTest("DeleteS3ScheduleAndCreateNfsSchedule", "Take a schedule backup of namespace in S3 backup location,delete the s3 schedule, create new schedule backup with NFS backup location or vice-versa", nil, 86099)
+		log.InfoD("Scheduling Applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		for i := 0; i < 5; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				appNamespaces = append(appNamespaces, namespace)
+				scheduledAppContexts = append(scheduledAppContexts, ctx)
+			}
+		}
+		log.Infof("The list of namespaces deployed are", appNamespaces)
+		providers = getProviders()
+	})
+
+	It("Take a schedule backup of namespace in S3 backup location, delete the s3 schedule, create new schedule backup with NFS backup location or vice-versa", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step("Adding cloud credential and backup locations", func() {
+			log.InfoD("Adding cloud credential and backup locations")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, provider := range providers {
+				s3CloudCredName = fmt.Sprintf("%s-%s-%v", "cloudcred", provider, RandomString(5))
```

Comment:
@snigdha-px Can this moved as one library API ?

Reply:
Sure @trenukarya-px .
Discussed this with Mithun. Since it is a common step in all the testcases, will take it up in future PR.